blockchain/blockchain_test.go

```diff
@@ -1644,3 +1659,400 @@ func TestBlockchain_WriteFullBlock(t *testing.T) {
 	require.NotNil(t, db[hex.EncodeToHex(getKey(storage.CANONICAL, common.EncodeUint64ToBytes(header.Number)))])
 	require.NotNil(t, db[hex.EncodeToHex(getKey(storage.RECEIPTS, header.Hash.Bytes()))])
 }
+
+func TestDiskUsageWriteBatchAndUpdate(t *testing.T) {
+	const (
+		checkInterval  = 10000 //milliseconds
+		numberOfBlocks = 100
+		blockTime      = 2000 //milliseconds
+	)
+
+	jsonFile, err := os.Open("testblock.json")
+	require.NoError(t, err)
+
+	byteValue, _ := io.ReadAll(jsonFile)
+
+	receiptsJSONFile, err := os.Open("receipts.json")
+	require.NoError(t, err)
+
+	receiptsByteVaulue, _ := io.ReadAll(receiptsJSONFile)
+
+	blockWriter(t, checkInterval, blockTime, numberOfBlocks, byteValue, receiptsByteVaulue)
+}
+
+func TestDiskUsageWriteBatchAndUpdateNoTimeout(t *testing.T) {
+	const (
+		checkInterval  = 15 // milliseconds
+		numberOfBlocks = 100
+		blockTime      = 0 //milliseconds
+	)
+
+	jsonFile, err := os.Open("testblock.json")
+	require.NoError(t, err)
+
+	byteValue, _ := io.ReadAll(jsonFile)
+
+	receiptsJSONFile, err := os.Open("receipts.json")
+	require.NoError(t, err)
+
+	receiptsByteVaulue, _ := io.ReadAll(receiptsJSONFile)
+
+	blockWriter(t, checkInterval, blockTime, numberOfBlocks, byteValue, receiptsByteVaulue)
+}
+
+func blockWriter(t *testing.T, checkInterval, blockTime, numberOfBlocks uint64, byteToRead []byte, receiptsBytesToRead []byte) {
+	t.Helper()
+
+	type blockCounter struct {
+		mu sync.RWMutex
+		x  uint64
+	}
+
+	counter := &blockCounter{x: 0}
+
+	quitChan := make(chan bool)
+
+	p, err := os.MkdirTemp("", "DiskUsageTest")
+	require.NoError(t, err)
+
+	db, err := leveldb.NewLevelDBStorage(
+		filepath.Join(p),
+		hclog.NewNullLogger(),
+	)
+	require.NoError(t, err)
+
+	block, err := CustomJSONBlockUnmarshall(byteToRead)
+	require.NoError(t, err)
+
+	receipts, err := CustomReceiptsUnmarshall(receiptsBytesToRead)
+	require.NoError(t, err)
+
+	dirSizeCheck := func() {
+		for {
+			select {
+			case <-quitChan:
+				return
+			default:
+				dirSizeValue, err := DirSize(p)
+
+				if err != nil {
+					t.Log(err)
+				}
+
+				counter.mu.RLock()
+
+				t.Logf("BLOCK: %d DIRSIZE IS: %d and average is:%.2f", counter.x, dirSizeValue, float64(dirSizeValue)/float64(counter.x))
+
+				counter.mu.RUnlock()
+
+				time.Sleep(time.Millisecond * time.Duration(checkInterval))
+			}
+		}
+	}
+
+	blockchain := &Blockchain{db: db}
+
+	require.NoError(t, os.MkdirAll(p, 0755))
+
+	require.NoError(t, err)
+
+	dirSizeBeforeBlocks, err := DirSize(p)
+	require.NoError(t, err)
+	t.Logf("DIRSIZE IS: %d", dirSizeBeforeBlocks)
+
+	go dirSizeCheck()
+
+	for i := 0; i < int(numberOfBlocks); i++ {
+		batchWriter := storage.NewBatchWriter(db)
+		block.Block.Header.Number = uint64(i)
```

Comment:
After this line, you need to change some fields in transactions so that the hashes differ (which is the correct use case).

What I would change is `To` and `Nonce` fields on txs, this would be enough to change the hash. Maybe we can randomize some other fields, like `GasPrice`, `Input` etc. This would make tests even more diverse.